### PR TITLE
Exceptions

### PR DESCRIPTION
**Pull request by ChrisAusema**
added all the exceptions

### CODE REVIEW

File: Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/CollectionNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package hestia.backend.exceptions.ServerExceptions;
+
+
+
```
Comment (Feikoritsema) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/CollectionNotFoundException.java]: Redundant white lines

File: Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/PluginNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package hestia.backend.exceptions.ServerExceptions;
+
+/**
+ * Created by chris on 28-5-2017.
```
Comment (Feikoritsema) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/PluginNotFoundException.java]: Not really needed + white lines

File: Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/ServerException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package hestia.backend.exceptions.ServerExceptions;
+
+
```
Comment (Feikoritsema) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/ServerException.java]: Again

File: Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/UndefinedServerException.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package hestia.backend.exceptions.ServerExceptions;
+
+/**
+ * Created by chris on 28-5-2017.
```
Comment (Feikoritsema) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/UndefinedServerException.java]: Same

**Review by Feikoritsema — CHANGES_REQUESTED** [review]
Overall looks fine, but some redundant white lines and comments at the top which are not needed.

File: Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/UndefinedServerException.java (hunk 4 above)
Comment (StefanEvanghelides) [reply Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/UndefinedServerException.java]: You can also delete the "Created by chris..."

**Review by StefanEvanghelides — APPROVED** [review]

**Review by StefanEvanghelides — APPROVED** [review]

**Review by StefanEvanghelides — CHANGES_REQUESTED** [review]
Some exceptions are randomly thrown

File: Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java [hunk 5]
```diff
@@ -44,11 +44,11 @@ public ServerCollectionsInteractor(NetworkHandler handler){
             JsonObject jsonObject = payload.getAsJsonObject();
             String error = jsonObject.get("error").getAsString();
             String message = jsonObject.get("message").getAsString();
-            throw new ComFaultException(error, message);
+            throw new DeviceNotFoundException(error, message);
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java:47]: The payload not being a json array does not mean that the device is not founf.

File: Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java [hunk 6]
```diff
@@ -64,42 +64,42 @@ public void addDevice(RequiredInfo info) throws IOException, ComFaultException {
             if(object.has("error")) {
                 String error = object.get("error").getAsString();
                 String message = object.get("message").getAsString();
-                throw new ComFaultException(error, message);
+                throw new DeviceNotFoundException(error, message);
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java:67]: The payload containing a error does not mean that the device was not found

File: Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java [hunk 7]
```diff
@@ -64,42 +64,42 @@ public void addDevice(RequiredInfo info) throws IOException, ComFaultException {
...
+    public void removeDevice(Device device) throws IOException, DeviceNotFoundException {
         String endpoint = "devices/" + device.getId();
         JsonElement payload = handler.DELETE(endpoint);
         if(payload != null && payload.isJsonObject()) {
             JsonObject jsonObject = payload.getAsJsonObject();
             if(jsonObject.has("error")) {
                 String error = jsonObject.get("error").getAsString();
                 String message = jsonObject.get("message").getAsString();
-                throw new ComFaultException(error, message);
+                throw new DeviceNotFoundException(error, message);
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java:80]: Same

File: Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java [hunk 8]
```diff
@@ -64,42 +64,42 @@ public void addDevice(RequiredInfo info) throws IOException, ComFaultException {
...
-    public RequiredInfo getRequiredInfo(String collection, String plugin) throws IOException, ComFaultException {
+    public RequiredInfo getRequiredInfo(String collection, String plugin) throws IOException, DeviceNotFoundException {
         JsonElement rawObject = handler.GET("plugins/" + collection + "/plugins/" + plugin);
         if (rawObject.isJsonObject()) {
             JsonObject object = rawObject.getAsJsonObject();
             if(object.has("error")) {
                 String error = object.get("error").getAsString();
                 String message = object.get("message").getAsString();
-                throw new ComFaultException(error, message);
+                throw new DeviceNotFoundException(error, message);
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java:102]: Same

File: Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java [hunk 9]
```diff
@@ -121,8 +121,8 @@ public RequiredInfo getRequiredInfo(String collection, String plugin) throws IOE
             }.getType());
             return list;
         } else if (element.getAsJsonObject().has("error")){
-            ComFaultException comFaultException=gson.fromJson(element,ComFaultException.class);
-            throw comFaultException;
+            DeviceNotFoundException deviceNotFoundException =gson.fromJson(element,DeviceNotFoundException.class);
+            throw deviceNotFoundException;
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/ServerCollectionsInteractor.java:125]: Same

File: Hestia/app/src/main/java/hestia/backend/models/Activator.java [hunk 10]
```diff
@@ -75,7 +75,7 @@ public void setState(ActivatorState state) throws IOException, ComFaultException
             if(result.has("error")) {
                 String error = result.get("error").getAsString();
                 String message = result.get("message").getAsString();
-                throw new ComFaultException(error, message);
+                throw new DeviceNotFoundException(error, message);
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/models/Activator.java:78]: Same

File: Hestia/app/src/main/java/hestia/backend/models/Device.java [hunk 11]
```diff
@@ -69,16 +69,16 @@ public String getName() {
...
         JsonObject object = new JsonObject();
         object.addProperty("name", name);
         JsonElement payload = handler.PUT(object, path);
         if(payload.isJsonObject() && payload.getAsJsonObject().has("error")){
             GsonBuilder gsonBuilder = new GsonBuilder();
             Gson gson = gsonBuilder.create();
-            ComFaultException comFaultException = gson.fromJson(payload, ComFaultException.class);
-            throw comFaultException;
+            DeviceNotFoundException deviceNotFoundException = gson.fromJson(payload, DeviceNotFoundException.class);
+            throw deviceNotFoundException;
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/models/Device.java:81]: Same

**Review by LarsHoldijk — CHANGES_REQUESTED** [review]
There was nothing done with the Undefined Server Exceptions. These need to be thrown.

**Review by FrankTN — CHANGES_REQUESTED** [review]
The code in its current state is not building on Travis, try running it locally to find the problems. Once they are resolved, run the tests locally and then push. In this case you can be fairly certain that the build will pass Travis.

**Comment by StefanEvanghelides** [comment]
I suggest we close this pull request. Will fix this part during the next sprint

**Pull request closed without merging**